chart/scripts/post-delete-hook.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+set -e
+
+namespaces="${NAMESPACES}"
+timeout="${TIMEOUT}"
+ignoreTimeoutError="${IGNORETIMEOUTERROR}"
+
+if [ -z "${namespaces}" ]; then
+  echo "No namespace is provided."
+  exit 1
+fi
+
+if [ -z "${timeout}" ]; then
+  echo "No timeout value is provided."
+  exit 1
+fi
+
+if [ -z "${ignoreTimeoutError}" ]; then
+  echo "No ignoreTimeoutError value is provided."
+  exit 1
+fi
+
+succeeded=()
+failed=()
+
+get_pod_count() {
+  kubectl get pods --selector app="${1}" -n "${2}" -o json | jq '.items | length'
+}
+
+echo "Uninstalling Rancher resources in the following namespaces: ${namespaces}"
+
+for namespace in ${namespaces}; do
+  for app in $(helm list -n "${namespace}" -q); do
+    if [[ ${app} =~ .crd$ ]]; then
+      echo "--- Skip the app [${app}] in the namespace [${namespace}]"
+      continue
+    fi
+    echo "--- Deleting the app [${app}] in the namespace [${namespace}]"
+    if [[ ! $(helm uninstall "${app}" -n "${namespace}") ]]; then
+      failed=("${failed[@]}" "${app}")
+      continue
+    fi
+    t=0
+    # by default, wait 120 seconds in total for an app to be uninstalled
+    while [[ $(get_pod_count "${app}" "${namespace}") != "0" ]] && [[ "${t}" -lt "${timeout}" ]]; do
+      echo "waiting 5 seconds for pods of [${app}] to be terminated ..."
+      sleep 5
+      t=$((t + 5))
+    done
+    if [[ $(get_pod_count "${app}" "${namespace}") != "0" ]]; then
```

Comment:
```suggestion
    if [[ "$(get_pod_count ${app} ${namespace})" -ne 0 ]]; then
```

Reply:
If the deletion is successful the first time around, this seems like an unnecessary second call to be made. Could we move the failure logic directly into the previous while statement?

e.g. 

```shell
while : ; do
    pod_count=$(get_pod_count ${app} ${namespace})
    if [[ "${pod_count}" -eq 0 ]]; then
        echo "successfully uninstalled [${app}] in the namespace [${namespace}]"
        succeeded=("${succeeded[@]}" "${app}")
        break
    fi
    if [[ "${t}" -ge "${timeout}" ]]; then
        echo "timeout uninstalling [${app}] in the namespace [${namespace}]"
        failed=("${failed[@]}" "${app}")
        break
    fi
    echo "waiting 5 seconds for pods of [${app}] to be terminated ..."
    sleep 5
    t=$((t + 5))
done
```

(note: `while : ` is bash for `while true`)

Reply:
Good point! The script is updated to use this `while` loop.